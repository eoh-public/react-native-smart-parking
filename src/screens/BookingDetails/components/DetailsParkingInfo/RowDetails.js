import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';

import Text from '../../../../commons/Text';
import { Colors } from '../../../../configs';

const RowDetails = memo(({ title = '', value = [], semibold = false }) => {
  return (
    <View style={styles.row}>
      <Text type={'Body'} color={Colors.Gray7}>
        {title}
      </Text>
      <View style={styles.col}>
        {value.map((item, index) => (
          <Text
            key={index}
            type={'Body'}
            color={semibold ? Colors.Gray8 : Colors.Gray7}
            semibold={semibold}
            style={index !== 0 && styles.rowInCol}
          >
            {item}
          </Text>
        ))}
      </View>
    </View>
  );
});

export default RowDetails;
const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 8,
  },
  col: {
    flexDirection: 'column',
  },
  rowInCol: {
    marginTop: 8,
  },
});
