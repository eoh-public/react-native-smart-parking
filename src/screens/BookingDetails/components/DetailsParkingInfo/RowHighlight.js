import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';
import { Colors } from '../../../../configs';
import Text from '../../../../commons/Text';

const RowHighlight = memo(({ title, value }) => {
  return (
    <View style={styles.row}>
      {title && (
        <Text bold type={'H4'} color={Colors.Gray9}>
          {title}
        </Text>
      )}
      {value && (
        <Text type={'H4'} color={Colors.Gray9}>
          {value}
        </Text>
      )}
    </View>
  );
});

export default RowHighlight;
const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 16,
  },
});
