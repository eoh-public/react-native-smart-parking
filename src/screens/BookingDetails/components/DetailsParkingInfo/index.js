import React, { memo } from 'react';
import { View, StyleSheet } from 'react-native';
import { t } from 'i18n-js';

import { formatMoney } from '../../../../utils/Utils';
import { TESTID } from '../../../../configs/Constants';

import RowDetails from './RowDetails';
import RowHighlight from './RowHighlight';

const DetailsParkingInfo = memo(
  ({
    id,
    arrive_at,
    leave_at,
    grand_total,
    service_fee,
    discount,
    total,
    payment_method,
    is_violated,
    plate_number,
    city,
    extend_fee,
    book_at,
    extend_at,
    pay_at,
  }) => {
    const timeFormat = 'LT - DD/MM/YYYY';
    const arrive_at_str = arrive_at && arrive_at.format(timeFormat);
    const leave_at_str = leave_at && leave_at.format(timeFormat);
    const book_at_str = book_at && book_at.format(timeFormat);
    const extend_at_formated = extend_at.map((item) => item.format(timeFormat));
    const pay_at_str = pay_at && pay_at.format(timeFormat);

    return (
      <View style={styles.container}>
        <View style={styles.content}>
          <RowHighlight title={t('booking_info')} />
          <RowDetails
            title={t('license_plate')}
            value={[`${plate_number || '--'}`]}
          />
          <RowDetails
            title={!is_violated ? t('arrive_at') : t('start_time')}
            value={[`${arrive_at_str || '--'}`]}
            testID={TESTID.ARRIVE_AT}
          />

          <RowDetails
            title={!is_violated ? t('leave_at') : t('end_time')}
            value={[`${leave_at_str || '--'}`]}
            testID={TESTID.LEAVE_AT}
          />
        </View>

        <View style={styles.bottomContent}>
          <RowHighlight title={t('parking_fee')} />
          <RowDetails
            testID={TESTID.NUMBER_OF_HOUR_PARKING}
            title={
              !is_violated
                ? t('total_parking_hours')
                : t('violating_parking_hours')
            }
            value={[`${formatMoney(total)}`]}
          />
          {!is_violated ? (
            <>
              {extend_at.length > 0 && (
                <RowDetails
                  title={t('extend_fee')}
                  value={[`${formatMoney(extend_fee)}`]}
                />
              )}
              <RowDetails
                title={t('service_fee')}
                value={[`${formatMoney(service_fee)}`]}
              />
              <RowDetails
                title={t('discount')}
                value={[`-${formatMoney(discount)}`]}
              />
            </>
          ) : (
            <>
              <RowDetails
                testID={TESTID.DETAIL_PARKING_INFO_VIOLATION_RATE}
                title={t('violation_rate')}
                value={city && [`x${city.violation_charge_rate}`]}
              />
              <RowDetails
                testID={TESTID.DETAIL_PARKING_INFO_VIOLATION_FEE}
                title={t('violation_fee')}
                value={city && [`${formatMoney(city.violation_charge_fee)}`]}
              />
            </>
          )}
          <RowHighlight
            title={t('text_total')}
            value={`${grand_total ? formatMoney(grand_total) : '--VND'}`}
            style={styles.styleTotal}
          />
        </View>

        <View style={styles.bottomContent}>
          <RowHighlight title={t('text_detail')} />
          <RowDetails title={t('booking_id')} value={[`#${id}`]} />
          <RowDetails
            title={t('payment_method')}
            value={[payment_method || '--']}
          />
          {!is_violated && (
            <RowDetails title={t('book_at')} value={[book_at_str || '--']} />
          )}

          <RowDetails
            testID={TESTID.PAY_AT}
            title={t('pay_at')}
            value={[pay_at_str || '--']}
          />
          {extend_at.length > 0 && (
            <RowDetails title={t('extend_at')} value={extend_at_formated} />
          )}
        </View>
      </View>
    );
  }
);

export default DetailsParkingInfo;

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
  },
  content: {
    paddingVertical: 16,
  },
  bottomContent: {
    paddingBottom: 16,
  },
  styleTotal: {
    fontSize: 14,
    lineHeight: 22,
  },
});
