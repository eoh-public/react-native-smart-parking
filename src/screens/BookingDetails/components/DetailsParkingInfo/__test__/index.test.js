import React from 'react';
import renderer, { act } from 'react-test-renderer';
import moment from 'moment';
import DetailsParkingInfo from '../index';

describe('Test DetailsParkingInfo', () => {
  const extend_at = [
    moment('2021-01-23T04:34:57.465029Z').utcOffset(0),
    moment('2021-01-23T04:34:32.204043Z').utcOffset(0),
  ];
  const book_at = moment('2021-01-23T03:34:57.465029Z').utcOffset(0);
  const grand_total = 10000;
  let tree;

  test('render details parking info', async () => {
    const bookingDetail = {
      extend_at: extend_at,
      book_at: book_at,
      pay_at: book_at,
      num_of_hour_parking: 1,
      grand_total: grand_total,
      extend_fee: grand_total,
      service_fee: 0,
      discount: 0,
      total: 20000,
      parking_id: 2,
      payment_method: 'word',
    };
    await act(async () => {
      tree = renderer.create(<DetailsParkingInfo {...bookingDetail} />);
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('render details parking info num of hour parking greater than 1', async () => {
    const bookingDetail = {
      extend_at: extend_at,
      book_at: book_at,
      num_of_hour_parking: 10,
      extend_fee: grand_total,
      service_fee: 0,
      discount: 0,
      total: 20000,
      parking_id: 2,
      payment_method: 'word',
    };
    await act(async () => {
      tree = renderer.create(<DetailsParkingInfo {...bookingDetail} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
