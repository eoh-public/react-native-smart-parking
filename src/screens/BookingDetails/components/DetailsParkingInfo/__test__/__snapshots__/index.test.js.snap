// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DetailsParkingInfo render details parking info 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Semibold",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      Chi tiết
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        1 giờ đỗ xe
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          20.000 đ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Phí gia hạn
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          10.000 đ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Phí dịch vụ
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          0 đ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Khuyến mãi
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          -0 đ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Tổng cộng
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          10.000 đ
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        ID đặt chỗ
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          #2
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Đặt chỗ lúc
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          3:34 AM - 23/01/2021
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Thanh toán lúc
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          3:34 AM - 23/01/2021
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Gia hạn lúc
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          4:34 AM - 23/01/2021
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
              },
            ]
          }
        >
          4:34 AM - 23/01/2021
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Hình thức thanh toán
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          word
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Test DetailsParkingInfo render details parking info num of hour parking greater than 1 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 16,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#E8E8E8",
        "borderBottomWidth": 1,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Semibold",
            "fontSize": 14,
            "lineHeight": 22,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      Chi tiết
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        10 giờ đỗ xe
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          20.000 đ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Phí gia hạn
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          10.000 đ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Phí dịch vụ
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          0 đ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Khuyến mãi
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          -0 đ
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Tổng cộng
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#595959",
                "fontFamily": "SFProDisplay-Semibold",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          0
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        ID đặt chỗ
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          #2
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Đặt chỗ lúc
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          3:34 AM - 23/01/2021
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Thanh toán lúc
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          Not paid
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Gia hạn lúc
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          4:34 AM - 23/01/2021
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {
                "marginTop": 8,
              },
            ]
          }
        >
          4:34 AM - 23/01/2021
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8C8C8C",
              "fontFamily": "SFProDisplay-Regular",
              "fontSize": 14,
              "lineHeight": 22,
              "textAlign": "left",
              "textDecorationLine": "none",
            },
            Object {},
          ]
        }
      >
        Hình thức thanh toán
      </Text>
      <View
        style={
          Object {
            "flexDirection": "column",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8C8C8C",
                "fontFamily": "SFProDisplay-Regular",
                "fontSize": 14,
                "lineHeight": 22,
                "textAlign": "left",
                "textDecorationLine": "none",
              },
              Object {},
            ]
          }
        >
          word
        </Text>
      </View>
    </View>
  </View>
</View>
`;
